### ISSUE TITLE
Diff object with status 'EQUAL' is missing original value

### ISSUE THREAD

**Issue by Aleeexx** [netilon/differify#18]
### What I have done
I diffed two arrays (see A and B at the bottom of this post) using `compareArraysInOrder: false`:
```ts
const diff = new Differy({
  mode: { object: "DIFF", array: "DIFF" },
  compareArraysInOrder: false,
}).compare(A, B)
```
and nearly everything works as expected:
* The _deleted one_ (`diff._.[0]`, with `phrase: "I was deleted"`) has `status: 'DELETED'` and its original value.
* The _last equal one_ (`diff._.[5]`, with `phrase "Help me with"`) has `status: 'EQUAL'` and its original (and current?) value as well.
* **But** the _first equal one_ (`diff._.[1]`) has `status: 'EQUAL'` but NOT its original value. Am I missing anything? Is this a bug?

### The output of the comparison:
```json
{
  "_": [
    {
      "original": { "id": 155, "phrase": "I was deleted" },
      "current": null,
      "status": "DELETED",
      "changes": 1
    },
    {
      "status": "EQUAL",
      "changes": 0
    },
    {
      "original": null,
      "current": { "id": 123, "phrase": "Was edited" },
      "status": "ADDED",
      "changes": 1
    },
    {
      "original": { "id": 123, "phrase": "Was edite" },
      "current": null,
      "status": "DELETED",
      "changes": 1
    },
    {
      "original": null,
      "current": { "id": 88, "phrase": "Was added in between" },
      "status": "ADDED",
      "changes": 1
    },
    {
      "original": { "id": 157, "phrase": "Help me with" },
      "current": { "id": 157, "phrase": "Help me with" },
      "status": "EQUAL",
      "changes": 0
    }
  ]
}
```

Array **A**
```ts
[
  {
    id: 155,
    phrase: "I was deleted",
  },
  {
    id: 156,
    phrase: "Can you help me with",
  },
  {
    id: 123,
    phrase: "Was edite",
  },
  {
    id: 157,
    phrase: "Help me with",
  },
]
```
Array **B**
```ts
[
  {
    id: 156,
    phrase: "Can you help me with",
  },
  {
    id: 123,
    phrase: "Was edited",
  },
  {
    id: 88,
    phrase: "Was added in between",
  },
  {
    id: 157,
    phrase: "Help me with",
  },
]
```

### Findings
If I move up the "broken object" (`{id: 156, phrase: "Can you help me with" }`) to the first position in the Array A (meaning swapping positions with A[0]) it works fine and the "bug" does not occur.
Even if I move up the added one (`{ id: 88, phrase: "Was added in between" }`) just before the "broken object" in the Array B, it works fine.
I seems that this occurs if the first element from Array A is deleted in Array B and the second one remained equal.

**Comment by fabian-orue** [comment]
Hi @Aleeexx , thank you for letting me know! and yes, it's a bug because all the results of "compare" method,  should return an "original" property (property of the first object in the comparison ) and a "current" property (current state of the property of the second object in the comparison) using the **DIFF** configuration. I will check it.

**Comment by fabian-orue** [comment]
Hi @Aleeexx , the issue you mentioned was fixed in the new version **3.2.3**. Regards!
